tower-http/src/services/fs/serve_dir.rs

```diff
@@ -254,12 +268,22 @@ impl<ReqBody> Service<Request<ReqBody>> for ServeDir {
                 full_path.set_extension(new_extension);
             }
 
+            let metadata = std::fs::metadata(&full_path)?;
```

Comment:
This should use [`tokio::fs::metadata`](https://docs.rs/tokio/latest/tokio/fs/fn.metadata.html) to not block the runtime. We already do some things with the metadata on `master` so if you rebase it would be relatively easy to fix.